### PR TITLE
add two node local net script

### PR DESCRIPTION
**Pull request by coriolinus**
It's non-trivial to run a multi-node local net. This PR adds a script which makes it a pushbutton operation.

### CODE REVIEW

File: scripts/two-node-local-net.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  # compute offsets
+  local port=$((30333+node_offset))
+  local rpc_port=$((9933+node_offset))
+  local ws_port=$((9944+node_offset))
+  node_offset++
+
+  # start the node
+  # TODO: do we need a custom chainspec? Probably.
+  "$polkadot" \
+    --chain polkadot \
```
Comment (coriolinus) [inline scripts/two-node-local-net.sh:62]: The intent is to run a polkadot net, identical to the polkadot software, but not actually connect to the real Polkadot network. I suspect that we need a custom chainspec here, but I don't yet know what will be precisely required.

File: scripts/two-node-local-net.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  mkfifo "$stdout"
+  mkfifo "$stderr"
+  node_pipes+=("$stdout")
+  node_pipes+=("$stderr")
+
+  # compute ports from offset
+  local port=$((30333+node_offset))
+  local rpc_port=$((9933+node_offset))
+  local ws_port=$((9944+node_offset))
+  node_offset=$((node_offset+1))
```
Comment (ordian) [inline scripts/two-node-local-net.sh:65]: I guess this will work only for up to 11 nodes, probably fine, but maybe worth documenting?

**Review by ordian — APPROVED** [review]
There is also [docker-compose](https://github.com/paritytech/polkadot/blob/master/docker/docker-compose-local.yml), but I guess the difference is that this one builds on HEAD instead of the latest released version?

I'm not well versed in bash enough in order tell you if there are better ways of doing this (maybe @s3krit is?), but it seems to work locally, which is what matters the most.

File: scripts/two-node-local-net.sh (hunk 2 above)
Comment (coriolinus) [reply scripts/two-node-local-net.sh:65]: You're right that it's limited to 11 nodes, but there's another, stricter limit: this only works properly if the node name is one of the default nodes: alice, bob, charlie, dave, eve, ferdie, one, two. We can't get to the 11 limit even by manually editing the script.

**Comment by coriolinus** [comment]
bot merge

**Comment by bkchr** [comment]
We already have something more advanced: https://github.com/shawntabrizi/polkadot-launch

**Comment by coriolinus** [comment]
That's interesting, and definitely more advanced, but in this case I think simplicity is a virtue. The point here is to be able to take any given state of the codebase and quickly and accurately answer the question: "Can it still produce blocks?"

**Pull request merged**